packages/flutter_test/lib/src/binding.dart

```diff
@@ -1424,8 +1444,8 @@ class LiveTestWidgetsFlutterBinding extends TestWidgetsFlutterBinding {
     switch (source) {
       case TestBindingEventSource.test:
         if (!renderView._pointers.containsKey(event.pointer)) {
-          assert(event.down);
-          renderView._pointers[event.pointer] = _LiveTestPointerRecord(event.pointer, event.position);
+          if (event.down)
```

Comment:
This `assert` doesn't take into consideration for multiple input events at the same time. This often happen for `PointerAddedEvent` and `PointerDownEvent`.

Reply:
> Is it true that we should only expect `PointerAddedEvent ` and `PointerDownEvent` here? 

For the test runs I did, yes. But I can say for sure. According to the documents I think `PointerEnterEvent` and `PointerScrollEvent` may also go here. This depends on runtime history of the input events and is hard to track the source.

Reply:
But since it's test, I would say I''ll still add assert and if people find something in the future they can add more.

Reply:
> Is it true that we should only expect `PointerAddedEvent ` and `PointerDownEvent` here? 

For the test runs I did, yes. But I can't say for sure. According to the documents I think `PointerEnterEvent` and `PointerScrollEvent` may also go here. This depends on runtime history of the input events and is hard to track the source. 

But since it's test, I would say I''ll still add assert and if people find something in the future they can add more.